shared/src/address/payload.rs

```diff
@@ -151,6 +151,17 @@ impl From<&Payload> for Protocol {
     }
 }
 
+impl TryFrom<Payload> for DelegatedAddress {
```

Comment:
Also implement by reference (not sure if necessary... but idiomatic).

Reply:
not really less or more idiomatic IMO, its just whether or not we are going to reuse payload after this call.